src/indexer/services/evmIndexer/evmEventParser.ts

```diff
@@ -0,0 +1,193 @@
+import BasicFeeHandlerContract from "@buildwithsygma/sygma-contracts/build/contracts/BasicFeeHandler.json"
+import { BigNumber } from "@ethersproject/bignumber"
+import {
+  Contract,
+  LogDescription,
+  Provider,
+  TransactionReceipt,
+  getBytes,
+  formatUnits,
+  AbiCoder,
+  hexlify,
+  Log,
+  BytesLike,
+  BigNumberish,
+} from "ethers"
+import { Domain, Resource, ResourceTypes } from "indexer/config"
+import { BridgeABI } from "../contract/constants"
+import { logger } from "../../../utils/logger"
+import { getERC20Contract } from "../contract"
+import {
+  DecodedDepositLog,
+  DecodedFailedHandlerExecution,
+  DecodedFeeCollectedLog,
+  DecodedLogs,
+  DecodedProposalExecutionLog,
+  EventType,
+  FeeHandlerType,
+} from "./evmTypes"
+
+export const nativeTokenAddress = "0x0000000000000000000000000000000000000000"
+
+export async function decodeLogs(provider: Provider, domain: Domain, logs: Log[], resourceMap: Map<string, Resource>): Promise<DecodedLogs> {
+  const decodedLogs: DecodedLogs = {
+    deposit: [],
+    proposalExecution: [],
+    errors: [],
+    feeCollected: [],
+  }
+  await Promise.all(
+    logs.map(async log => {
+      const blockUnixTimestamp = (await provider.getBlock(log.blockNumber))?.timestamp || 0
+      const contractData = domain.feeHandlers.filter(handler => handler.address == log.address)
+
+      let decodedLog: LogDescription | null = null
+      if (contractData[0]?.type == FeeHandlerType.BASIC) {
```

Comment:
Move this logic to some getDecodedLog function.
It should throw if it is null.

Reply:
done